spinoso-time/src/time/tzrs/offset/error.rs

```diff
@@ -0,0 +1,62 @@
+use core::fmt;
+use std::error;
+use std::str::Utf8Error;
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum OffsetError {
+    TzStringError(TzStringError),
+    OutOfRangeError(OutOfRangeError),
+}
+
+impl error::Error for OffsetError {}
+
+impl fmt::Display for OffsetError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OffsetError::TzStringError(error) => error.fmt(f),
+            OffsetError::OutOfRangeError(error) => error.fmt(f),
+        }
+    }
+}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TzStringError(pub(crate) String);
+
+impl fmt::Display for TzStringError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.write_str(&self.0)
+    }
+}
+impl error::Error for TzStringError {}
+
+#[allow(clippy::module_name_repetitions)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct OutOfRangeError;
+
+impl fmt::Display for OutOfRangeError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "utc_offset out of range")
```

Comment:
I'm in two minds about providing the string, or keeping it defaulted. The limitation of keeping it defaulted like this is that it's hard to inject what value was incorrect (e.g. in TzStringError it would be nice to error with the value that was provided).

The reason I put the defaulted values in like this was specifically because matching the error in the test was giving me some hassle, and I just wanted to get the test cases working first.

Do you think it's worth changing `TzStringError` tto provide the provided value?

Reply:
I think for now this is fine and we can defer this decision to later.